locust_tasks/locustfile.py

```diff
@@ -176,15 +229,121 @@ class UserBehavior(TaskSet):
     """
     
     tasks = {
-        LaunchEQ
+        LaunchEQ: 100,
+        LaunchEQInvalidUAC: 0,
+        LaunchEQwithAddressCorrection: 0,
+        request_new_code: 0,
+        launch_web_chat: 0
     }
 
 
 class WebsiteUser(HttpLocust):
     task_set = UserBehavior
     wait_time = between(2, 10)
 
-
     def setup(self):
         setup()
 
+
+#
+# This function should be called after each page transition as it aims to aggressively check that:
+#   - The current page is the expected page.
+#   - The actual http response status matches the expected status.
+#   - Optionally verifies that key content exists on the current page.
+#
+# In the event of failure it:
+#   - Reports key debugging information, such as step ID & the UAC, to aid with debugging.
+#   - For the error log it records the failure message and key content of the current page.
+#   - Aborts the current task.
+#
+def verify_response(id, task, resp, expected_status, expected_page, expected_content=''):
+    # print ('In verify_response(%s). Expected:%3d actual:%3d expected_page:%s' % (id, expected_status, resp.status_code, expected_page))
+    # print ('  URL:%s' % (resp.url))
+    # print ('  status:%d' % (resp.status_code))
+    # print ('  Expected page title:%s' % (expected_page.title))
+
+    # Sanity check for missing response 
+    if (resp.text in (None, '')):
+        failure_message = 'Expected to be on the ' + expected_page.name + ' page but got an empty response!'
+        report_failure(id, resp, task, failure_message, '')
+
+    # Page check
+    current_page = identify_page(id, task, resp, resp.text)
+    if (current_page != expected_page):
+        failure_message = 'On wrong page. Expected to be on ' + expected_page.name + ' page but am on ' + current_page.name + ' page.'
+        page_extract = extract_key_page_content(id, task, resp, current_page)
+        report_failure(id, resp, task, failure_message, page_extract)        
+    
+    # Status check
+    if (expected_status != resp.status_code):
+        failure_message = 'Status mismatch. Expected ' + str(expected_status) + ' but was ' + str(resp.status_code) + '.'
+        page_extract = extract_key_page_content(id, task, resp, current_page)
+        report_failure(id, resp, task, failure_message, page_extract)        
+    
+    # Content verification
+    if (expected_content not in (None, '')):
```

Comment:
just use python truthy/falsy:
```
if expected_content:
  doSomething
```

Same applies to other tests for emptiness.

Reply:
done